### PR TITLE
Update login design page

### PR DESCRIPTION
**Pull request by bamarni**
For the `dcos-oidc-auth0` login provider ID, this removes the need
to copy-paste the token between the UI / CLI by leveraging a
local webserver and the `redirect_uri` parameter.

https://developer.okta.com/blog/2018/07/16/oauth-2-command-line#the-trick-to-oauth-20-on-the-command-line
https://jira.mesosphere.com/browse/DCOS-51535

### CODE REVIEW

File: design/login.md [hunk 1]
```diff
@@ -116,3 +117,39 @@ Each login provider has one of these 5 login methods associated to it (`client-m
...
+message:
+
+``` console
+If your browser didn't open, please follow this link:
+
+    http://my-cluster.example.com/login?redirect_uri=http://localhost:8080&dcos_cli_csrf_token=g6qAFJeHUz3OViQHPPnwkCSjo3BVZSn4QiqmtrlqElo=
+```
+
+On successful login, our [Auth0 universal login page](https://github.com/mesosphere/auth0-ui) is
+configured to make a `GET` request to the `redirect_uri`, once it has validated that it refers to a `localhost` URL, with 2 parameters:
```
Comment (bamarni) [inline design/login.md]: @brandonc @juliangieseke Here I didn't put the details yet on how to make this request from auth0.com to localhost, any recommendation?

File: design/login.md [hunk 2]
```diff
@@ -116,3 +117,46 @@ Each login provider has one of these 5 login methods associated to it (`client-m
...
+
+For example:
+
+    http://localhost:8080?token=myLoginToken123&csrf=g6qAFJeHUz3OViQHPPnwkCSjo3BVZSn4QiqmtrlqElo=
+
+The Auth0 universal login page uses [fetch()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch)
+in order to perform the request. As this is a cross-domain request, the local web server
+is configured to support CORS and will only accept requests from the `https://dcos.auth0.com` origin.
+
+Once the request is done, the local web server can retrieve the token and continue the login flow.
```
Comment (jongiddy) [inline design/login.md]: Add that the web server must check that the CSRF token returned in the redirect URI matches the one sent.

**Comment by bamarni** [comment]
Addressed Jon's comment in the last commit, should be good to go now! thanks for the reviews

File: design/login.md [hunk 3]
```diff
@@ -116,3 +117,48 @@ Each login provider has one of these 5 login methods associated to it (`client-m
...
+For example:
+
+    http://localhost:8080?token=myLoginToken123&csrf=g6qAFJeHUz3OViQHPPnwkCSjo3BVZSn4QiqmtrlqElo=
+
+The Auth0 universal login page uses [fetch()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch)
+in order to perform the request. As this is a cross-domain request, the local web server
+is configured to support CORS and will only accept requests from the `https://dcos.auth0.com` origin.
+
+Once the request is done, the local web server can retrieve the token, verify that it matches with
+the token sent originally and continue the login flow.
```
Comment (jongiddy) [inline design/login.md]: Sorry to be a pain, but I think we should be clear about the login token and CSRF token here. The login token is what we're trying to get, but the CSRF token is the one we need to check matches.

File: design/login.md [hunk 4]
```diff
@@ -116,3 +117,48 @@ Each login provider has one of these 5 login methods associated to it (`client-m
...
+The user sees a successful login message on the webpage, indicating that they can go back to their terminal.
+
+However, if this request fails (eg. the CLI runs on a remote machine), the login page falls back
+to printing the token in a modal box, asking the user to copy-paste it to their terminal.
+The CLI will read it from stdin, verify that it matches with the token sent originally and continue
+the login flow.
```
Comment (jongiddy) [inline design/login.md]: I don't think verifying a match on the CSRF token in the manual fallback is required.

File: design/login.md (hunk 3 above)
Comment (bamarni) [reply design/login.md]: Indeed, I was confused myself! Updated now

**Pull request merged**